project_event/tests/test_project_event.py

```diff
@@ -46,3 +46,14 @@ def test_030_onchange_partner_id(self):
         self.project_1.partner_id = self.partner_tag2.id
         self.project_1._onchange_partner_id()
         self.assertEqual(self.project_1.client_type.name, 'Type 3')
+
+    def test_031_duplicate_activities(self):
```

Comment:
You are very explicit renaming the methods:         self.write_main_task(vals) ->
        self.write_children(vals) -> self.write_activty_children(vals), thats good, you could put a more minighful and descriptive name to this test.